File: Aditya-Pratap/Sorting/881.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        n = len(people)
        count = 0
        i, j = 0, n - 1
        while i <= j:
            if people[i] + people[j] <= limit:
                count += 1
                i += 1
                j -= 1
            else:
                count += 1
                j -= 1
        return count

File: [Manya]/Leetcode/Sorting/881.java
package Sorting;

import java.util.Arrays;

class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int start = 0;
        int end = people.length-1;
        int boats = 0;

        while(start <= end){
            if(people[start] + people[end] <= limit){
                boats++;
                start++;
                end--;
            }
            else{
                end--;
                boats++;
            }
            
        }
        return boats;
    }
}
